tests/utilities/test_apply_func.py

```diff
@@ -31,12 +32,80 @@ class Feature:
         input_ids: torch.Tensor
         segment_ids: np.ndarray
 
+        def __eq__(self, o: object) -> bool:
```

Comment:
noob q: why did you need to implement all the `__eq__` methods?

Reply:
They are not strictly necessary, but they avoid repeatedly writing the same or slightly varied checks for each occurrence of one of the dataclasses in the test. A single `==` works for each assertion.

I'll also note that dataclasses are automatically provided a default `__eq__`, but the defaults won't work here since `torch.Tensor` overrides `__eq__` to return tensors instead of bool.